presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java

```diff
@@ -96,6 +97,16 @@ public void completionEvent(QueryInfo queryInfo)
                 }
             }
 
+            TaskInfo task = null;
```

Comment:
This code seems duplicated. How about creating a method `getFailedTask(queryInfo)`.